File: 168.excel表列名称.ts
/*
 * @lc app=leetcode.cn id=168 lang=typescript
 *
 * [168] Excel表列名称
 */

// @lc code=start
function convertToTitle(columnNumber: number): string {
    let res = '';
    while (columnNumber) {
        const diff = (columnNumber - 1) % 26;
        res = getChar(diff) + res;
        columnNumber = (columnNumber - diff - 1) / 26;
    }
    return res;
}

const code = 'A'.charCodeAt(0);

function getChar(diff: number): string {
    return String.fromCharCode(code + diff);
}
// @lc code=end

File: 168.excel表列名称.py
#
# @lc app=leetcode.cn id=168 lang=python3
#
# [168] Excel表列名称
#

# @lc code=start
class Solution:
    def convertToTitle(self, columnNumber: int) -> str:
        res = ""
        while columnNumber:
            diff = (columnNumber - 1) % 26
            res = getChar(diff) + res
            columnNumber = (columnNumber - diff - 1) // 26

        return res


code = ord("A")


def getChar(diff: int):
    return chr(code + diff)


# @lc code=end
